content/yubikey-login.md

```diff
@@ -15,98 +15,94 @@ tableOfContents: true
 
-Pay close attention when copying/pasting the commands above.  The *challenge-123456* and *alice-123456* needs to match whatever your output is.**
+Pay close attention when copying/pasting the commands above.  The *challenge-123456* and *alice-123456* needs to match the both the output from the `ykpamcfg` command and the final file needs to match the name of your user name and serial.**
 
 ## Configure Plugable Authentication Modules
 
 **Before making any changes to the files listed below, I highly recommend backing up each file, and having a sudo/root session open in case you need to roll-back.**
 
-You need to add the following line to each of the files listed below:
+    sudo dpkg-reconfigure libpam-yubico
+
+You will want to change the "Parameters for Yubico PAM:" to be:
+
+    mode=challenge-response debug chalresp_path=/var/yubico
+
+![dpkg-reconfigure-pg1](/images/yubikey-login/dpkg-reconfigure-pg1.png)
+![dpkg-reconfigure-pg2](/images/yubikey-login/dpkg-reconfigure-pg2.png)
+
+If you want to only want to add the Yubikey as sufficient, you will need to change the second setting in the following line in ```/etc/pam.d/common-auth```:
 
     auth sufficient pam_yubico.so mode=challenge-response chalresp_path=/var/yubico
 
-This configures your system to accept your Yubikey as an acceptable login alternative.  In other words, you can login without a password, just plug the key into a USB port. **You can still use your password to login with this setup**.
+This configures your system to accept your Yubikey as an acceptable login alternative.  
 
-If you want your Yubiky to be **required to login**, then you need to change *sufficient* to *required*.  For example:
+If you want your Yubiky to be **required to login**, then you need to change *sufficient* to *required* (which is the default).  For example:
 
     auth required pam_yubico.so mode=challenge-response chalresp_path=/var/yubico
 
-There a a number of other options for PAM.  Check the man pages for more information.  Also, the [Yubico PAM module](https://developers.yubico.com/yubico-pam/) page has a pretty good breakdown.
+There a a number of other options for PAM and ways to configure that to your liking.  Check the [man pages for more information](https://manpages.ubuntu.com/manpages/jammy/en/man5/pam.d.5.html).  Also, the [Yubico PAM module](https://developers.yubico.com/yubico-pam/) page has a pretty good breakdown.
 
 - /etc/pam.d/common-auth
 - /etc/pam.d/login
 - /etc/pam.d/gdm-password
 
-### /etc/pam.d/common-auth
-
-At a minimum you need to modify this file.
+## Debugging and Testing your PAM configuration
 
-Add the auth line to the top of this file (after the comment block).
+You will want to keep a root terminal logged in while setting this up to make sure you can back out any changes that do not allow you to login.  In the initial setup, the parameters to setup libpam-yubico included debug message. "Sudo" on the command line will now show debug output when run.
```

Comment:
>make sure you can back out any changes

Should probably change "back out any changes" to "reverse any changes" for clarity?